src/Game/Scripting/Commands.cs

```diff
@@ -24,845 +24,1069 @@
-            Interpreter.RegisterCommandHandler("chivalryheal", UnimplementedCommand);
-            Interpreter.RegisterCommandHandler("moveitemoffset", UnimplementedCommand);
-            Interpreter.RegisterCommandHandler("movetype", UnimplementedCommand);
-            Interpreter.RegisterCommandHandler("movetypeoffset", UnimplementedCommand);
-            Interpreter.RegisterCommandHandler("togglemounted", UnimplementedCommand);
-            Interpreter.RegisterCommandHandler("equipwand", UnimplementedCommand);
-            Interpreter.RegisterCommandHandler("buy", UnimplementedCommand);
-            Interpreter.RegisterCommandHandler("sell", UnimplementedCommand);
-            Interpreter.RegisterCommandHandler("clearbuy", UnimplementedCommand);
-            Interpreter.RegisterCommandHandler("clearsell", UnimplementedCommand);
-            Interpreter.RegisterCommandHandler("organizer", UnimplementedCommand);
-            Interpreter.RegisterCommandHandler("counter", UnimplementedCommand);
-            Interpreter.RegisterCommandHandler("replygump", UnimplementedCommand);
-            Interpreter.RegisterCommandHandler("closegump", UnimplementedCommand);
-            Interpreter.RegisterCommandHandler("cleartargetqueue", UnimplementedCommand);
-            Interpreter.RegisterCommandHandler("autotargetlast", UnimplementedCommand);
-            Interpreter.RegisterCommandHandler("autotargetself", UnimplementedCommand);
-            Interpreter.RegisterCommandHandler("autotargetobject", UnimplementedCommand);
-            Interpreter.RegisterCommandHandler("autotargettype", UnimplementedCommand);
-            Interpreter.RegisterCommandHandler("autotargettile", UnimplementedCommand);
-            Interpreter.RegisterCommandHandler("autotargettileoffset", UnimplementedCommand);
-            Interpreter.RegisterCommandHandler("autotargettilerelative", UnimplementedCommand);
-            Interpreter.RegisterCommandHandler("autotargetghost", UnimplementedCommand);
-            Interpreter.RegisterCommandHandler("autotargetground", UnimplementedCommand);
-            Interpreter.RegisterCommandHandler("cancelautotarget", UnimplementedCommand);
-            Interpreter.RegisterCommandHandler("getenemy", UnimplementedCommand);
-            Interpreter.RegisterCommandHandler("getfriend", UnimplementedCommand);
+            DateTime lastExecuted = DateTime.UtcNow.AddHours(-2.0); // 2 hours ago
+            foreach(string key in commandKeywords)
+            {
+                if (_lastExecutedRegistry.ContainsKey(key) && _lastExecutedRegistry[key] > lastExecuted)
+                    lastExecuted = _lastExecutedRegistry[key];
+            }
+            return lastExecuted;
         }
 
-        private static bool UnimplementedCommand(string command, Argument[] args, bool quiet, bool force)
+        // Build execution
+        public CommandExecution(Command command, ParameterList parameters, bool quiet, bool force)
```

Comment:
Would it be simpler to combine Command and CommandExection into a single class? You can still only build the ParameterList right before calling the command handler.